back-nest/src/sсrapers/scraper-telemart.service.ts

```diff
@@ -1,15 +1,23 @@
 import { Injectable, Logger } from '@nestjs/common';
 import puppeteer from 'puppeteer';
+import { randomUUID } from 'crypto';
 
 import { Sources } from './models/sources';
 import { RedisService } from '../redis/redis.service';
+import { ProductService } from '../products/products.services';
 import { ScrapedProduct } from './models/scraped-product.model';
+import { ScraperService } from './models/scraper-service.model';
+import { S3Service } from '../s3/s3.service';
 
 @Injectable()
-export class ScraperTelemartService {
+export class ScraperTelemartService implements ScraperService {
   private readonly logger = new Logger(ScraperTelemartService.name);
 
-  constructor(private readonly redisService: RedisService) {}
+  constructor(
+    private readonly redisService: RedisService,
+    private readonly s3Service: S3Service,
+    private readonly productService: ProductService,
+  ) {}
 
   async scrapeTelemart(): Promise<ScrapedProduct[]> {
```

Comment:
Same naming issue as in rozetka service
```suggestion
  async scrapeProducts(): Promise<ScrapedProduct[]> {
```